src/main/java/io/xdb/core/process/BasicClientProcessOptions.java

```diff
@@ -0,0 +1,22 @@
+package io.xdb.core.process;
+
+import io.xdb.gen.models.AsyncStateConfig;
+import java.util.Optional;
+import lombok.Getter;
+
+@Getter
+public class BasicClientProcessOptions {
+
+    private final Optional<ProcessOptions> processOptionsOptional;
+    private final AsyncStateConfig startStateConfig;
+
+    public BasicClientProcessOptions(final ProcessOptions processOptions, final AsyncStateConfig startStateConfig) {
+        if (processOptions == null) {
```

Comment:
Nit: I think it can be just Optional.ofNullable(processOptions)

Reply:
Good learning!